arq/worker.py

```diff
@@ -459,14 +485,14 @@ async def job_failed(exc: Exception) -> None:
             try:
                 async with async_timeout.timeout(timeout_s):
                     result = await function.coroutine(ctx, *args, **kwargs)
-            except Exception as e:
+            except (Exception, asyncio.CancelledError) as e:
                 exc_extra = getattr(e, 'extra', None)
                 if callable(exc_extra):
                     exc_extra = exc_extra()
                 raise
             else:
                 result_str = '' if result is None else truncate(repr(result))
-        except Exception as e:
+        except (Exception, asyncio.CancelledError) as e:
```

Comment:
this is good, but it's important to note that it fixes current behaviour and will mean jobs are retried correctly upon restarting, not just by default when the `in_progress` key expires.